### PR TITLE
Update clang-newest configuration to use clang 10

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

File: libs/affinity/include/hpx/affinity/detail/partlit.hpp [hunk 1]
```diff
@@ -194,7 +194,7 @@ namespace hpx { namespace threads { namespace detail {
 
     private:
         // silence MSVC warning C4512: assignment operator could not be generated
-        partial_literal_string& operator=(partial_literal_string const&);
+        partial_literal_string& operator=(partial_literal_string const&) = default;
```
Comment (hkaiser) [inline libs/affinity/include/hpx/affinity/detail/partlit.hpp:197]: This might have to be `=delete` if anything. I'll investigate.

**Comment by msimberg** [comment]
test

File: libs/affinity/include/hpx/affinity/detail/partlit.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/affinity/include/hpx/affinity/detail/partlit.hpp]: I removed this declaration altogether as recent versions of the compiler don't generate the bogus warning anymore.

Comment (msimberg) [reply libs/affinity/include/hpx/affinity/detail/partlit.hpp]: Thank you!

**Pull request merged**